src/Illuminate/Support/Str.php

```diff
@@ -818,6 +818,17 @@ public static function ucfirst($string)
         return static::upper(static::substr($string, 0, 1)).static::substr($string, 1);
     }
 
+    /**
+     * Encodes the string to a URL-query-safe string.
+     *
+     * @param  $string
+     * @return string
+     */
+    public static function urlQuery($string)
+    {
+        return urlencode(trim($string));
```

Comment:
I don't think the trimming makes sense here either. What's the reason to not allow whitespace around the string. I think this method in the core is probably unnecessary. Just define a macro in your app that does what you need, or just call urlencode.